src/Bicep.LangServer/Providers/AzureContainerRegistriesProvider.cs

```diff
@@ -29,19 +30,25 @@ public class AzureContainerRegistryNamesProvider : IAzureContainerRegistryNamesP
 | where type == ""microsoft.containerregistry/registries""
 | project properties[""loginServer""]";
 
-        public AzureContainerRegistryNamesProvider(IConfigurationManager configurationManager, ITokenCredentialFactory tokenCredentialFactory)
+        public AzureContainerRegistriesProvider(IConfigurationManager configurationManager, ITokenCredentialFactory tokenCredentialFactory)
         {
             this.configurationManager = configurationManager;
             this.tokenCredentialFactory = tokenCredentialFactory;
         }
 
-        public async Task<IEnumerable<string>> GetRegistryNames(Uri templateUri)
+        public async Task<IEnumerable<string>> GetRegistryUris(Uri templateUri, CancellationToken cancellationToken)
```

Comment:
Instead of using `Task<IEnumerable<string>>` and returning a list, can we use `IAsyncEnumerable<string>` and `yield return` to lazily load the result?

Reply:
Considered that, but it won't make any difference - completion won't finish until everything is assembled together.

Reply:
I mean, I could still do it if you want...

Reply:
Makes sense. I think it's just an optional optimization for the code, but it won't make any difference in terms of user experience.

Reply:
I decided it made sense to pass that decision on to the client of IAzureContainerRegistriesProvider, so added it.  Learn how to test it was interesting...   So I learned some things. :-)

Reply:
5a95d6b4539095e85e52c1c56f8e4fef6696f397